core/src/test/java/com/linecorp/armeria/common/logging/DefaultRequestLogTest.java

```diff
@@ -73,4 +86,58 @@ public void rpcFailure_endResponseDifferentCause() {
         log.endResponse(error2);
         assertThat(log.responseCause()).isSameAs(error2);
     }
+
+    @Test
+    public void addChild() {
+        final DefaultRequestLog child = new DefaultRequestLog(ctx);
+        log.addChild(child);
+        child.startRequest(channel, SessionProtocol.H2C, "/examlple.com");
+        final long childStartTime = child.requestStartTimeMillis();
+        assertThat(log.requestStartTimeMillis()).isBetween(childStartTime, childStartTime + 300); // 300 buffer
+        assertThat(log.channel()).isSameAs(channel);
+        assertThat(log.sessionProtocol()).isSameAs(SessionProtocol.H2C);
+        assertThat(log.host()).isEqualTo("/examlple.com");
+
+        child.serializationFormat(SerializationFormat.NONE);
+        assertThat(log.serializationFormat()).isSameAs(SerializationFormat.NONE);
+
+        final HttpHeaders foo = HttpHeaders.of(AsciiString.of("foo"), "foo");
+        child.requestHeaders(foo);
+        assertThat(log.requestHeaders()).isSameAs(foo);
+
+        final String requestContent = "baz";
+        final String rawRequestContent = "bax";
+
+        child.requestContent(requestContent, rawRequestContent);
+        assertThat(log.requestContent()).isSameAs(requestContent);
+        assertThat(log.rawRequestContent()).isSameAs(rawRequestContent);
+
+        child.endRequest();
+        assertThat(log.requestDurationNanos()).isNotZero();
+
+        // response-side log are propagated when RequestLogBuilder.endResponseWithLastChild() is invoked
+        child.startResponse();
+        assertThatThrownBy(() -> log.responseStartTimeMillis())
+                .isExactlyInstanceOf(RequestLogAvailabilityException.class);
+
+        final HttpHeaders bar = HttpHeaders.of(AsciiString.of("bar"), "bar");
+        child.responseHeaders(bar);
+        assertThatThrownBy(() -> log.responseHeaders())
+                .isExactlyInstanceOf(RequestLogAvailabilityException.class);
+
+        log.endResponseWithLastChild();
+        assertThat(log.responseStartTimeMillis()).isEqualTo(child.responseStartTimeMillis());
+        assertThat(log.responseHeaders()).isSameAs(bar);
+
+        final String responseContent = "baz1";
+        final String rawResponseContent = "bax1";
+        child.responseContent(responseContent, rawResponseContent);
+        assertThat(log.responseContent()).isSameAs(responseContent);
+        assertThat(log.rawResponseContent()).isSameAs(rawResponseContent);
+
+        child.endResponse(new AnticipatedException("Oops!"));
+        final long childTotalDurationMillis = TimeUnit.NANOSECONDS.toMillis(child.totalDurationNanos());
+        assertThat(TimeUnit.NANOSECONDS.toMillis(log.totalDurationNanos()))
+                .isBetween(Math.max(0, childTotalDurationMillis - 300), childTotalDurationMillis); // 300 buffer
```

Comment:
- 300 buffer -> Could you rephrase?
- Could you extract 300 into a variable? It's used twice.
- How about increasing to 1000 for less flakiness?